### PR TITLE
add mask2former algo

### PR DESCRIPTION
**Pull request by haiasd**
- add mask2former algo
- support panopitc pipeline
- add segment predictor

### CODE REVIEW

File: configs/segmentation/mask2former/mask2former_r50_8xb2_e50_instance.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        num_points=12554,
+        no_object_weight=0.1,
+        oversample_ratio=3.0,
+        importance_sample_ratio=0.75,
+    ),
+    test_cfg=dict(
+        instance_on=True,
+        max_per_image=100,
+    ),
+    pretrained='https://download.pytorch.org/models/resnet50-19c8e357.pth',
```
Comment (wenmengzhou) [inline configs/segmentation/mask2former/mask2former_r50_8xb2_e50_instance.py:74]: @tuofeilunhifi check this, we should set  backbone.pretrained=True to load default model

File: docs/source/model_zoo_seg.md [hunk 2]
```diff
@@ -14,3 +14,15 @@ Pretrained on **Pascal VOC 2012 + Aug**.
 | Algorithm  | Config                                                       | mIoU | Download                                                     |
 | ---------- | ------------------------------------------------------------ | ------------------------ | ------------------------------------------------------------ |
 | upernet_r50 | [upernet_r50_512x512_8xb4_60e_voc12aug](https://github.com/alibaba/EasyCV/tree/master/configs/segmentation/upernet/upernet_r50_512x512_8xb4_60e_voc12aug.py) | 76.59               | [model](http://pai-vision-data-hz.oss-cn-zhangjiakou.aliyuncs.com/EasyCV/modelzoo/segmentation/upernet_r50/epoch_60.pth) - [log](http://pai-vision-data-hz.oss-cn-zhangjiakou.aliyuncs.com/EasyCV/modelzoo/segmentation/upernet_r50/20220706_114712.log.json) |
+
+## Mask2former
+
+### Instance Segmentation on COCO
```
Comment (wenmengzhou) [inline docs/source/model_zoo_seg.md:20]: also add a single readme.md to introduce mask2former in configs/segmentation/mask2former/

can be done in future PR

File: easycv/datasets/detection/data_sources/coco_panoptic.py [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+
+        if 'pan_results' in results:
+            pan_results = results['pan_results']
+            pan_json_results = self._pan2json(pan_results)
+            result_files['panoptic'] = f'{self.outfile_prefix}.panoptic.json'
+            mmcv.dump(pan_json_results, result_files['panoptic'])
+
+        return result_files
+
+    def get_gt_json(self, result_files):
```
Comment (wenmengzhou) [inline easycv/datasets/detection/data_sources/coco_panoptic.py:424]: add docstr

File: easycv/datasets/detection/pipelines/mm_transforms.py [hunk 4]
```diff
@@ -793,6 +799,44 @@ def __repr__(self):
         return repr_str
 
 
+@PIPELINES.register_module()
+class MMSegRescale:
+    """Rescale semantic segmentation maps.
```
Comment (wenmengzhou) [inline easycv/datasets/detection/pipelines/mm_transforms.py:804]: 合并到Resize

File: easycv/hooks/yolox_lr_hook.py [hunk 5]
```diff
@@ -4,7 +4,7 @@
                                           annealing_cos)
 
 
-@HOOKS.register_module()
+@HOOKS.register_module(force=True)
```
Comment (wenmengzhou) [inline easycv/hooks/yolox_lr_hook.py:7]: @Cathy0908  check this, any compatibility problem with mmdet?

File: easycv/models/segmentation/heads/mask2former_head.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+
+@HEADS.register_module()
+class Mask2FormerHead(nn.Module):
+
+    def __init__(
+        self,
+        pixel_decoder,
+        transformer_decoder,
+        *,
```
Comment (wenmengzhou) [inline easycv/models/segmentation/heads/mask2former_head.py:15]: what does * mean?

File: easycv/models/segmentation/heads/mask2former_head.py [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+            ignore_value (int, optional): category id to be ignored during training. Defaults to -1.
+            transformer_in_feature (str, optional): nput feature name to the transformer_predictor, only support multi_scale_pixel_decoder now. Defaults to 'multi_scale_pixel_decoder'.
+        """
+        super().__init__()
+
+        self.ignore_value = ignore_value
+        self.common_stride = 4
+        self.loss_weight = loss_weight
+        self.pixel_decoder = MSDeformAttnPixelDecoder(**pixel_decoder)
+        self.predictor = MultiScaleMaskedTransformerDecoder(
```
Comment (wenmengzhou) [inline easycv/models/segmentation/heads/mask2former_head.py:40]: self.decoder

File: easycv/models/segmentation/utils/matcher.py [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+        'nc,mc->nm', neg, (1 - targets))
+    return loss / hw
+
+
+batch_sigmoid_ce_loss_jit = torch.jit.script(
+    batch_sigmoid_ce_loss)  # type: torch.jit.ScriptModule
+
+
+# Modified from https://github.com/facebookresearch/detr/blob/master/models/matcher.py
+class HungarianMatcher(nn.Module):
```
Comment (wenmengzhou) [inline easycv/models/segmentation/utils/matcher.py:56]: @tuofeilunhifi can we reuse these module with DETR series

File: easycv/models/segmentation/heads/mask2former_head.py (hunk 6 above)
Comment (haiasd) [reply easycv/models/segmentation/heads/mask2former_head.py:15]: after * must be keyword parameter

File: easycv/models/segmentation/utils/matcher.py (hunk 8 above)
Comment (haiasd) [reply easycv/models/segmentation/utils/matcher.py:56]: This is quite different from the original

File: easycv/models/segmentation/heads/mask2former_head.py (hunk 7 above)
Comment (haiasd) [reply easycv/models/segmentation/heads/mask2former_head.py]: done

File: easycv/datasets/detection/pipelines/mm_transforms.py (hunk 4 above)
Comment (haiasd) [reply easycv/datasets/detection/pipelines/mm_transforms.py]: done

File: easycv/datasets/detection/data_sources/coco_panoptic.py (hunk 3 above)
Comment (haiasd) [reply easycv/datasets/detection/data_sources/coco_panoptic.py:424]: done

File: configs/segmentation/mask2former/mask2former_r50_8xb2_e50_instance.py (hunk 1 above)
Comment (haiasd) [reply configs/segmentation/mask2former/mask2former_r50_8xb2_e50_instance.py]: done

File: easycv/hooks/yolox_lr_hook.py (hunk 5 above)
Comment (haiasd) [reply easycv/hooks/yolox_lr_hook.py:7]: This function is the same as mmdet, so it doesn't matter which one to use

File: easycv/models/segmentation/heads/mask2former_head.py (hunk 7 above)
Comment (haiasd) [reply easycv/models/segmentation/heads/mask2former_head.py:40]: change name will cause previous model load problem

**Review by wenmengzhou — APPROVED** [review]

File: easycv/models/segmentation/heads/mask2former_head.py (hunk 6 above)
Comment (haiasd) [reply easycv/models/segmentation/heads/mask2former_head.py]: done

**Pull request merged**